packages/core/src/render3/instructions/shared.ts

```diff
@@ -112,10 +112,37 @@ function refreshContentQueries(tView: TView, lView: LView): void {
   }
 }
 
+/**
+ * Different modes of traversing the logical view tree during change detection.
+ *
+ *
+ * The change detection traversal algorithm switches between these modes based on various
+ * conditions.
+ */
+const enum ChangeDetectionMode {
+  /**
+   * In `Global` mode, `Dirty` and `CheckAlways` views are refreshed as well as views with the
+   * `RefreshTransplantedView` flag.
+   */
+  Global,
+  /**
+   * In `Targeted` mode, only views with the `RefreshTransplantedView`
+   * flag are refreshed.
+   */
+  Targeted,
+  /**
+   * Used when refreshing a view to force a refresh of its embedded views. This mode
+   * refreshes views without taking into account their LView flags. i.e. non-dirty OnPush components
+   * will be refreshed in this mode.
+   */
+  BugToForceRefreshAndIgnoreViewFlags
```

Comment:
nit: I think it'd be great to add some extra info into the comment section to mention why we have this mode and if there are followup steps we'd need to take here (like getting rid of this mode in v17?).
```suggestion
   * refreshes views without taking into account their LView flags, i.e. non-dirty OnPush components
   * will be refreshed in this mode.
   */
  BugToForceRefreshAndIgnoreViewFlags
```